src/neuroconv/nwbconverter.py

```diff
@@ -184,3 +183,42 @@ def run_conversion(
                 )
 
         return nwbfile_out
+
+
+class NWBConverterPipe(NWBConverter):
+    """Takes pre-initialized interfaces as arguments to build an NWBConverter class"""
```

Comment:
Structure of the class definitely looks better, question now is what name to choose... other options might be `DynamicNWBConverter` (from https://github.com/catalystneuro/nwb-conversion-tools/pull/110/files#diff-91a2f17a5200d62cf54bb3dc0051c2df6c96650f247faf2bc514327211840919R138), `NWBConverterContainer` (reflecting treatment of acting as a bucket/wrapper of already created objects), or any others we can think of to consider...

Any thoughts @bendichter?

Reply:
My argument for something along the lines of Pipe is that a pipeline is a common metaphor in the scientific python community for a series of processing operations. See [scikit-learn](https://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html) and [pandas](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.pipe.html).

I think @CodyCBakerPhD mention that there were some overlap with some objects in Datalad.

I don't think the word dynamic is specific enough. As I see it, it makes reference to the python data model (as classes are created dynamically, meta-programming, etc.) but I don't see what is the utility of communicating that to the end users. 

This is meant to be a function that end users access frequently and I think it should be named by their function. Maybe there is something better than pipe?

Reply:
100% agree that a NWBConverter, as a class structure, is analogous to a data pipeline in the sense that you specify which data interfaces to use (which themselves are pipelines of which types of conversion protocols to use) but my point is that is true even of the base NWBConverter class as well - this new subclass does not change or introduce that base behavior, so my philosophy for choosing a name for a child class like this would be to find the one-two simplest word(s) that minimally differentiates it from the parent and summarizes that difference

Reply:
I am not sure I agree with that naming philosophy. I want to have a name that is useful and easy to remember for users. I don't think the goal of the name here should be to describe how this object is different from the NWBConverter (parent) as I don't think that that information is useful to the user.

I still think that `InterfacePipeline` is a good name as it refers to both the atomic pieces of the conversion (the interfaces) and it states what it does with them. I changed it here to `NWBConverterPipe` because of the ambiguity concerns that you mentioned, that is, the mention of `NWBConverter` was accidental as its goal was to disambiguate.

Other possibilities are just calling it  `Pipe` like scikit-learn or `ConversionPipe` if we want to be more specific.

Reply:
Changed it to `ConverterPipe` to avoid a possible confusion with the `NWBConverter`. Other possible name is `InterfacePipe`.

Reply:
`ConverterPipe` is fine